### PR TITLE
FEATURE: BB-2215 Add custom definition syntax highlighting

### PR DESCRIPTION
**Pull request by scavnickyj**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [ ] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [ ] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [ ] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] Successful `extended test - examples`
- [ ] Successful `extended test - storybook`
- [ ] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer:
- gdc-dashboards:

# Related Jira tasks

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7538/ : SUCCESS in 7m 13s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3630/ : SUCCESS in 3s

**Comment by scavnickyj** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3954/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3631/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3312/ : UNSTABLE in 12m 04s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7540/ : SUCCESS in 11m 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3634/ : SUCCESS in 3s

**Comment by scavnickyj** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3955/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3635/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3314/ : SUCCESS in 10m 16s

**Comment by scavnickyj** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1628.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3636/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3315/ : SUCCESS in 18m 54s

File: src/components/measureNumberFormat/syntaxHighlighting/SyntaxHighlighting.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
+// (C) 2020 GoodData Corporation
+import * as React from "react";
+import { UnControlled as CodeMirrorInput } from "react-codemirror2";
+import * as CodeMirror from "codemirror";
+import "codemirror/lib/codemirror.css";
+import "codemirror/addon/mode/simple";
+import "codemirror/addon/edit/matchbrackets";
+
+const measureFormatMode = (CodeMirror as any).defineSimpleMode("numberFormat", {
```
Comment (tomassatinsky) [inline src/components/measureNumberFormat/syntaxHighlighting/SyntaxHighlighting.tsx:9]: The comments seems to be from an example and are not related to our implementation so I would remove them
see https://github.com/google/comid/blob/master/example/mode/simplemode.html

File: src/components/measureNumberFormat/syntaxHighlighting/SyntaxHighlighting.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    editor: CodeMirror.Editor;
+}
+
+export interface ISyntaxHighlightingProps {
+    value: string;
+    onChangeHandler?: (value: string) => void;
+    className?: string;
+}
+
+export const SyntaxHighlighting: React.FC<ISyntaxHighlightingProps> = props => {
```
Comment (tomassatinsky) [inline src/components/measureNumberFormat/syntaxHighlighting/SyntaxHighlighting.tsx:60]: Consider naming it "MeasureFormatInput"

File: src/components/measureNumberFormat/syntaxHighlighting/tests/SyntaxHighlighting.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
+// (C) 2020 GoodData Corporation
```
Comment (tomassatinsky) [inline src/components/measureNumberFormat/syntaxHighlighting/tests/SyntaxHighlighting.spec.tsx:1]: I'm OK if you remove the test :)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7542/ : SUCCESS in 5m 34s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3637/ : SUCCESS in 5s

**Comment by scavnickyj** [comment]
extended test - storybook - filter "Measure Number Format"

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3957/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3639/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3316/ : UNSTABLE in 12m 42s

**Comment by scavnickyj** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3958/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3641/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3318/ : UNSTABLE in 11m 16s

**Comment by scavnickyj** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3959/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3650/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3322/ : UNSTABLE in 16m 52s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7557/ : FAILURE in 1m 58s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3665/ : SUCCESS in 5s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7558/ : SUCCESS in 6m 40s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3666/ : SUCCESS in 5s

**Comment by scavnickyj** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3966/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3669/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3332/ : UNSTABLE in 11m 11s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7561/ : SUCCESS in 6m 13s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3671/ : SUCCESS in 5s

**Comment by scavnickyj** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3967/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3673/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3334/ : SUCCESS in 11m 11s

**Comment by scavnickyj** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1628.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3674/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3335/ : SUCCESS in 17m 37s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7562/ : SUCCESS in 2m 54s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3675/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3971/ : SUCCESS in 11m 37s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-fossa-scanning-zuul-el7/50/ : SUCCESS in 5m 48s (non-voting)
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7567/ : SUCCESS in 3m 16s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1826/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2700/ : SUCCESS in 15m 47s

**Pull request merged**